flaml/tune/online_trial.py

```diff
@@ -0,0 +1,416 @@
+                 is_champion: Optional[bool] = False,
+                 is_checked_under_current_champion: Optional[bool] = True,
+                 custom_trial_name: Optional[str] = 'vw_mae_clipped',
+                 trial_id: Optional[str] = None,
+                 cb_coef: Optional[float] = None,
+                 ):
+        """Constructor
+
+        Args:
+            config (dict): the config of the trial (note that the config is a set because the hyperparameters are )
+            min_resource_lease (float): the minimum resource lease
+            metric (str): the loss metric
+            is_champion (bool): indicates whether the trial is the current champion or not
+            is_checked_under_current_champion (bool): indicates whether this trials has been paused under the current champion
+            trial_id (str): id of the trial (if None, it will be generated in the constructor)
+
+        """
+        self.trial_id = self._config_to_id(config) if trial_id is None else trial_id
+        logger.info('Create trial with trial_id: %s', self.trial_id)
+        super().__init__(config, min_resource_lease, is_champion, is_checked_under_current_champion,
+                         custom_trial_name, self.trial_id)
+        # variables that are needed during online training
+        self._metric = metric
+        self._y_min_observed = None
+        self._y_max_observed = None
+        self.model = None   # model is None until the config is scheduled to run
+        self.result = None
+        # application dependent info
+        self._dim = None
+        self._namespace_set = None
+        self.trainable_func = self.MODEL_CLASS
+        self.cb_coef = cb_coef
+
+    @staticmethod
+    def _config_to_id(config):
+        """Generate an id for the provided config
+        """
+        # sort config keys
+        sorted_k_list = sorted(list(config.keys()))
+        config_id_full = ''
+        for key in sorted_k_list:
+            v = config[key]
+            if isinstance(v, set):
+                config_id = ''
+                key_list = sorted(v)
+                for k in key_list:
+                    config_id = config_id + '_' + str(k)
+            else:
+                config_id = str(v)
+            config_id_full = config_id_full + config_id
+        return config_id_full
+
+    def initialize_vw_model(self, vw_example):
+        """Initialize a vw model using the trainable_fuc
+        """
+        # get the dimensionality of the feature according to the namespace configuration
+        self._vw_config_tuned = self.config.copy()
+        # ensure the feature interaction config is a list (required by VW)
+        if VWOnlineTrial.interactions_config_key in self._vw_config_tuned.keys():
+            self._vw_config_tuned[VWOnlineTrial.interactions_config_key] \
+                = list(self._vw_config_tuned[VWOnlineTrial.interactions_config_key])
+        self._namespace_set = self.config[VWOnlineTrial.interactions_config_key]
```

Comment:
this line indicates line 282 is not needed. 
put this line before line 283 and simplify line 284